core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -187,8 +187,7 @@ protected void cleanUncommitted(Set<ManifestFile> committed) {
           deleteFile(manifest.path());
         }
       }
-
-      this.newManifests = committedNewManifests;
+      this.newManifests = null;
```

Comment:
I doubt that this is the correct fix. I don't think that `newManifests` should be modified at all. What was the issue with leaving them as-is?

Reply:
No issue when using a transaction, but without a transaction the commit passes and leaves the table in an invalid (the manifest file is deleted and can't be read) state. I changed the test to use a transaction and reverted this change.  But this leaves the non-transaction use case unsolved for now.

Reply:
The test has been replaced and a dedicated transaction one added.